packages/blocks/src/api/utils.js

```diff
@@ -121,3 +123,105 @@ export function normalizeBlockType( blockTypeOrName ) {
 
 	return blockTypeOrName;
 }
+
+/**
+ * Get the label for the block, usually this is either the block title,
+ * or the value of the block's `label` function when that's specified.
+ *
+ * @param {Object} blockType  The block type.
+ * @param {Object} attributes The values of the block's attributes.
+ * @param {Object} context    The intended use for the label.
+ *
+ * @return {string} The block label.
+ */
+export function getBlockLabel( blockType, attributes, context = 'visual' ) {
+	const {
+		__experimentalLabel: getLabel,
+		title,
+	} = blockType;
+
+	const label = getLabel && getLabel( attributes, { context } );
+
+	if ( ! label ) {
+		return title;
+	}
+
+	// Strip any HTML (i.e. RichText formatting) before returning.
+	return stripHTML( label );
+}
+
+/**
+ * Get a label for the block for use by screenreaders, this is more descriptive
+ * than the visual label and includes the block title and the value of the
+ * `getLabel` function if it's specified.
+ *
+ * @param {Object}  blockType              The block type.
+ * @param {Object}  attributes             The values of the block's attributes.
+ * @param {?number} position               The position of the block in the block list.
+ * @param {string}  [direction='vertical'] The direction of the block layout.
+ *
+ * @return {string} The block label.
+ */
+export function getAccessibleBlockLabel( blockType, attributes, position, direction = 'vertical' ) {
```

Comment:
We've moving slowly away from functions that take blockType, and attributes in favor of the whole block object. I also feel like position and direction are not necessary and can be retrieved from state if this is becomes more a selector or a component instead of a global function. cc @aduth

Reply:
> We've moving slowly away from functions that take blockType, and attributes in favor of the whole block object.

Right. It already raises the question: What if I wanted the label to change depending on the inner blocks? How would that be implemented today with this interface?

Also, there's some inconsistency between this utility and the others in the file; where most all of the other utilities will accept _either_ a block type (object) _or_ a block name (string).

Reply:
I mentioned it above in another comment, but I'm now looking at moving this out of the `blocks` package into `block-editor` as selectors.

the api would be `getAccessibleBlockLabel( clientId, direction )` and the selector implementation could be responsible for grabbing any other data.